spec/util/object_storage/miq_swift_storage_spec.rb

```diff
@@ -0,0 +1,61 @@
+require "util/object_storage/miq_swift_storage"
+
+describe MiqSwiftStorage do
+  let(:object_storage) { described_class.new(:uri => uri, :username => 'user', :password => 'pass') }
+
+  context "using a uri without query parameters" do
+    let(:uri) { "swift://foo.com/abc/def" }
+
+    it "#initialize sets the container_name" do
+      container_name = object_storage.container_name
+      expect(container_name).to eq("abc/def")
+    end
+
+    it "#uri_to_object_path returns a new object path" do
+      result = object_storage.uri_to_object_path(uri)
+      expect(result).to eq("def")
+    end
+  end
+
+  describe "#auth_url (private)" do
+    context "with non-ssl security protocol" do
+      let(:uri) { "swift://foo.com:5678/abc/def?region=region&api_version=v3&security_protocol=non-ssl" }
```

Comment:
Not sure if it is worth considering, but do we want to add a tests to confirm that both the `path` and `query_string` are not part of the generated `auth_url`?

Reply:
Oh derp, yeah, that is right.  I was more saying that the path didn't equal `/abc/def` from the `uri` being passed in, but yes, that sounds right.  Sorry for the confusion.